app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,41 @@
+class VotesController < ApplicationController
+  def upvote
```

Comment:
This entire method is pretty long--it should be refactored to call some separate methods on the `Vote` model that do your logic for you.